File: TTS/app/src/main/java/com/example/tts/MainActivity.java
package com.example.tts;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.speech.tts.TextToSpeech;
import android.view.View;

import java.util.Locale;

public class MainActivity extends AppCompatActivity implements TextToSpeech.OnInitListener {

    private TextToSpeech tts;

    private boolean ttsLoaded = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        tts = new TextToSpeech(this, this);
    }

    @Override
    public void onInit(int status){
        if(status == TextToSpeech.SUCCESS){
            tts.setLanguage(Locale.getDefault());
            ttsLoaded = true;
        }
    }


    public void falar (View view){
        if(ttsLoaded){
            String S = "Olá mundo";
            tts.speak(S, TextToSpeech.QUEUE_FLUSH, null, null);
        }
    }
}

File: PPDM/app/src/main/java/com/example/ppdm/MainActivity.kt
package com.example.ppdm

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    protected void OnStart(){
        super.OnStart();
        Log.d("Ciclo de ida Activivty")
    }
}